service/view_building_state.hh

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ *
+ */
+
+/*
+ * SPDX-License-Identifier: (LicenseRef-ScyllaDB-Source-Available-1.0 and Apache-2.0)
+ */
+
+#pragma once
+
+#include "db/view/view_build_status.hh"
+#include "locator/host_id.hh"
+#include "locator/tablets.hh"
+#include "utils/UUID.hh"
+#include <fmt/base.h>
+
+namespace service {
+
+namespace view_building {
+
+// Holds information about tasks for some tablet_replica.
+//
+// build_range      ->  Build a token_range owned by tablet_id, from base_id to view_id. 
+//                      Multiple `build_range` tasks are executed together 
+//                      if their `base_id` and `tid` are the same.
+// process_staging  ->  Register all staging sstables owned by the tablet_replica 
+//                      to the view_update_generator and wait until all of the staging sstables 
+//                      are processed (view updates are generated and 
+//                      the sstables are moved to normal directory).
+struct view_building_task {
+    enum class task_type {
+        build_range,
+        process_staging,
+    };
+
+    // When a task is created, it starts with `IDLE` state.
+    // Then, the view building coordinator will decide to do the task and it will
+    // set the state to `STARTED`.
+    // When a task is finished or aborted, the entry is removed.
+    enum class task_state {
+        idle,
+        started,
+    };
+    utils::UUID id;
+    task_type type;
+    task_state state;
+
+    table_id base_id;
+    table_id view_id; // is default value when `type == task_type::process_staging`
+    locator::tablet_replica replica;
+    locator::tablet_id tid;
+};
+
+using task_map = std::map<utils::UUID, view_building_task>;
+
+struct replica_tasks {
+    std::map<table_id, task_map> view_tasks;    // owned by a particular view
+    task_map staging_tasks;                     // owned by whole base table
+};
+
+using base_table_tasks = std::map<locator::tablet_replica, replica_tasks>;
+using building_tasks = std::map<table_id, base_table_tasks>;
+
+// Represents view building state (only for tablet-based views).
+// Stores tasks for view building coordinator and currently processed base table.
+struct view_building_state {
```

Comment:
Please introduce constructors for these structs. Right now it is possible to create them via direct-list-initialization (`view_building_state state { arg1, arg2, ... }`) but it is very easy to forget to specify one of the fields at the end with this syntax.